modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -1387,6 +1392,20 @@ private void validateCommon(IgniteConfiguration cfg) {
         A.ensure(cfg.getNetworkSendRetryCount() > 0, "cfg.getNetworkSendRetryCount() > 0");
     }
 
+    /**
+     * Check whether UTF-8 is the default character encoding.
+     * Differing character encodings across cluster may lead to erratic behavior.
+     */
+    private void checkFileEncoding() {
+        String encodingDisplayName = Charset.defaultCharset().displayName(Locale.ENGLISH);
+
+        if (!"UTF-8".equals(encodingDisplayName)) {
```

Comment:
Probably we can warn user only if we detect different encoding are applied cluster-wide. I'm not sure we should warn here if we did not validate overall settings.